### ISSUE TITLE
Sync fails if pipe character "|" exists in bookmark title

### ISSUE THREAD

**Issue by vogtgbr** [floccusaddon/floccus#1800]
### Which version of floccus are you using?

5.4.0

### How many bookmarks do you have, roughly?

3k

### Are you using other means to sync bookmarks in parallel to floccus?

No

### Sync method

WebDAV

### Which browser are you using? In case you are using the phone App, specify the Android or iOS version and device please.

Firefox

### Describe the Bug

If there exists a pipe character 
`|`
in the title of a bookmark like so:
`  <bookmark href="https://www.thingiverse.com/thing:6542198" id="5457">`
`    <title>One-Handed Soldering Tool 2.0 | Effortless Soldering With Only One Hand by RoboticWorx - Thingiverse</title>`
`  </bookmark>`
and a sync runs and puts this in the .xbel file on the WebDav, all consecutive syncs from any device will fail with
`Content-Length header of network response exceeds response Body. | `  
Interestingly the error message contains the character that causes the trouble.

### Expected Behavior

Pipe characters in bookmark titles should not cause trouble. ;)

### To Reproduce

1.) Create a bookmark with a title that contains character | (pipe)
2.) Let it sync upward so that all devices can see this bookmark
3.) Sync again to get the error message.

### Debug log provided

- [ ] I have provided a debug log file

**Comment by marcelklehr** [comment]
Hi @vogtgbr 

I believe you are mistaken about the pipe character being the problem. I have multiple bookmarks with pipes in the title that sync fine.

The pipe character is intentionally part of the error message: 
![image](https://github.com/user-attachments/assets/318cb1bc-c2a6-4338-91bc-667f69540ae4)

The error message in your case indicates that your server has a problem: Content-Length header of network response exceeds response Body.

**Comment by vogtgbr** [comment]
Hi Marcel,
thats really strange. I manually edited the bookmarks.xbel file on the WebDav server and changed only this single character from | to -. That did the trick, syncing worked again without problems after that. However, you are right, I discovered now that I have several titles with a pipe character myself, so this can't be the (only) reason for the problem.
I'll try to do more research on this matter and if I figure out more, I'll be happy if I may tell you about it. For now, I guess you can close this issue.
Thanks for your time!

**Comment by marcelklehr** [comment]
Nah, let's leave this open until we figure it out :)

**Comment by marcelklehr** [comment]
>  I manually edited the bookmarks.xbel file on the WebDav server and changed only this single character from | to -.

It may have been just the edit that made it work?

**Comment by vogtgbr** [comment]
Jep, I didn't do anything else and I don't know why it started working again. Maybe the time stamp of bookmarks.xbel was the reason? It must have changed when I edited it.

**Comment by shirocko** [comment]
Hi. I got the same error and can reproduce it again and again, but for me the error occurs at the tab synchronisation.
The following error occurs, when I have the startpage of t-online.de open as a tab.
![image](https://github.com/user-attachments/assets/485ea653-66e4-4208-9156-072dcd4569f3)
The title of the page always has a "|" in the name and that seems to block or stop the synchronisation.
If I navigate to a sub page of t-online.de, when the sync is working fine.
![image](https://github.com/user-attachments/assets/b25dba2e-8ee0-424c-ac24-10420d1524fc)

**Comment by marcelklehr** [comment]
Hi @shirocko 
That's curious. Are you also using WebDAV to sync? Can you check your server logs for the 500 error?

**Comment by shirocko** [comment]
Hi @marcelklehr 
I am not using webdav, but I use a self hosted nextcloud instance based on docker and in the logs I only see the same error, as in the debug logs from the floccus addon.

Output debug.log addon:

> Syncing failed with E019: HTTP status 500. Failed PUT request. Check your server configuration and log.

Output apache log of nextcloud docker container:

> bookmarksync [03/Jan/2025:20:49:36 +0000] "PUT /index.php/apps/bookmarks/public/rest/v2/bookmark/3763 HTTP/1.1" 500 1930 "-" "Mozilla..."

**Comment by marcelklehr** [comment]
@shirocko Please open a new ticket for this, as this ticket is about webdav. Also, can you check the nextcloud log for errors?
